point.go

```diff
@@ -338,3 +338,20 @@ func (p Point) ToWKT() string {
 func (p Point) String() string {
 	return fmt.Sprintf("POINT(%g %g)", p[0], p[1])
 }
+
+// Round returns a point which coordinates are rounded
+// Factor defaults to 1.0e5
+func (p *Point) Round(factor ...int) *Point {
+	f := 1.0e5
+	if len(factor) != 0 {
+		f = float64(factor[0])
+	}
+
+	res := Point{}
```

Comment:
sorry to be so slow on review. I will be better going forward.

This library modifies points inplace (for better or worse). so it should just modify `p` and return it vs creating a new one. 

See the Add and Subtract methods for examples: https://github.com/paulmach/go.geo/blob/master/point.go#L217-L230

I eventually found that to be annoying in practice and wrote what I think is a more idiomatic library https://github.com/paulmach/orb